dags/reporting-dag.py

```diff
@@ -46,21 +45,11 @@ def set_max_task_instance(ti, **kwargs):
     ti.xcom_push(key='MAX_TASK_START', value=str(max_task_start[0]))
 
 
-default_args = {
-    'owner': 'airflow',
-    'depends_on_past': False,
-    'email_on_failure': False,
-    'email_on_retry': False,
-    'retries': 1,
-    'retry_delay': timedelta(minutes=5)
-}
-
 with DAG('reporting_dag',
          start_date=datetime(2019, 1, 1),
          max_active_runs=3,
          schedule_interval=None,
-         default_args=default_args,
-         # catchup=False # enable if you don't want historical dag runs to run
+         catchup=False
          ) as dag:
     t0 = PostgresOperator(
```

Comment:
Would be nice to have the variable names match the operator names for clarity. Additionally, the `sql` arg here can be templated according to the comment on `template_searchpath` above as `sql={% include 'rpt.sql' %}`.

The TaskGroups below can each just be their own DAG, with the `t0` task repeated in each DAG. This is nice because it gives us more DAGs on the registry, but also because it makes each DAG more resilient, i.e. if one TaskGroup fails, the whole DAG will fail, when that's probably not correct behavior.

A next step beyond that, which I don't think you need to implement but could make a good blog post, is further templating the TaskGroups --> individual DAGs --> DAG Factory, as they all have relatively similar form.